src/common/reducers/actions.js

```diff
@@ -1417,14 +1425,136 @@ export function downloadForge(instanceName) {
         });
       });
 
-      await fse.remove(path.join(_getTempPath(state), modloader[2]));
+      await fse.remove(path.join(_getTempPath(state), loader?.loaderVersion));
     }
 
     await fse.remove(tempInstaller);
   };
 }
 
-export function processManifest(instanceName) {
+export function processFTBManifest(instanceName) {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const { manifest } = _getCurrentDownloadItem(state);
+    // const concurrency = state.settings.concurrentDownloads;
+    const instancesPath = _getInstancesPath(state);
+    const instancePath = path.join(instancesPath, instanceName);
+
+    const { files } = manifest;
+    const concurrency = state.settings.concurrentDownloads;
+
+    let modManifests = [];
+
+    const updatePercentage = downloaded => {
+      dispatch(updateDownloadProgress((downloaded * 100) / files.length));
+    };
+
+    dispatch(updateDownloadStatus(instanceName, 'Downloading FTB files...'));
+    await downloadInstanceFiles(
+      files,
+      updatePercentage,
+      undefined,
+      instancePath
+    );
+
+    await pMap(
+      files,
+      async item => {
+        let ok = false;
+        let tries = 0;
+        /* eslint-disable no-await-in-loop */
+        do {
+          tries += 1;
+          if (tries !== 1) {
+            await new Promise(resolve => setTimeout(resolve, 5000));
+          }
+          try {
+            const ext = item.name.split(/\.(?=[^.]+$)/)[1];
+            const isJarFile = ext === 'jar';
+            if (isJarFile) {
+              const filePath = path.join(instancePath, item.path, item.name);
+
+              const hash = await getFileMurmurHash2(filePath);
+              const { data } = await getAddonsByFingerprint([hash]);
```

Comment:
This is highly inefficient because you need to make 1 API request for each mod. You should first calculate all the hashes and then only make a bunch of requests with an array of hashes to minimize the requests.